Running GK fails for repo sqlalchemy-exasol

Running GK fails for repo sqlalchemy-exasol:

```console
 ✘ user@host ~/github-keeper $ go run . configure-repo "sqlalchemy-exasol"

https://github.com/exasol/sqlalchemy-exasol
panic: unsupported type float64

goroutine 1 [running]:
github.com/exasol/github-keeper/cmd.WorkflowDefinitionParser.replaceSpecificParameterInJobName({}, {0xc0000c62a0, 0x20}, {0x822360, 0xc000526f98}, 0x4e328d)
        /home/nic/Projects/github-keeper/cmd/workflowDefinitionParser.go:140 +0x16e
github.com/exasol/github-keeper/cmd.WorkflowDefinitionParser.replaceParametersInJobName({}, {0xc0000c62a0, 0x19}, 0x8e8f6a, {0xc000534ab0, 0x3, 0x3}, 0x0)
        /home/nic/Projects/github-keeper/cmd/workflowDefinitionParser.go:116 +0x1e5
github.com/exasol/github-keeper/cmd.WorkflowDefinitionParser.fillJobNameParametersForMatrixBuild({}, {0xc00009c0b8, 0xc000542f10}, {0xc0000c62a0, 0x6f})
        /home/nic/Projects/github-keeper/cmd/workflowDefinitionParser.go:79 +0x325
github.com/exasol/github-keeper/cmd.getJobNames(0xc0005430b0, {})
        /home/nic/Projects/github-keeper/cmd/workflowDefinitionParser.go:60 +0x125
github.com/exasol/github-keeper/cmd.WorkflowDefinitionParser.ParseWorkflowDefinition({}, {0xc000571000, 0xfbe})
        /home/nic/Projects/github-keeper/cmd/workflowDefinitionParser.go:35 +0xca
github.com/exasol/github-keeper/cmd.BranchProtectionVerifier.getChecksForWorkflowContent({{0x7ffe4bc4322c, 0x0}, 0xc000188000}, {0xc000571000, 0xfbe}, 0xc000542030)
        /home/nic/Projects/github-keeper/cmd/branchProtection.go:230 +0x167
github.com/exasol/github-keeper/cmd.BranchProtectionVerifier.getChecksForWorkflow({{0x7ffe4bc4322c, 0xc000188088}, 0xc000188000}, 0xc000542030)
        /home/nic/Projects/github-keeper/cmd/branchProtection.go:225 +0x85
github.com/exasol/github-keeper/cmd.BranchProtectionVerifier.getRequiredChecks({{0x7ffe4bc4322c, 0xc00018a190}, 0xc000188000}, 0x0)
        /home/nic/Projects/github-keeper/cmd/branchProtection.go:204 +0x1ad
github.com/exasol/github-keeper/cmd.BranchProtectionVerifier.createProtectionRequest({{0x7ffe4bc4322c, 0x98e4f0}, 0xc000188000}, 0x2a)
        /home/nic/Projects/github-keeper/cmd/branchProtection.go:164 +0xa5
github.com/exasol/github-keeper/cmd.BranchProtectionVerifier.CheckIfBranchProtectionIsApplied({{0x7ffe4bc4322c, 0xc000010018}, 0xc000188000}, 0x1e)
        /home/nic/Projects/github-keeper/cmd/branchProtection.go:55 +0x17c
github.com/exasol/github-keeper/cmd.glob..func1(0xca5240, {0xc000063040, 0x1, 0x1})
        /home/nic/Projects/github-keeper/cmd/configureRepo.go:29 +0x15e
github.com/spf13/cobra.(*Command).execute(0xca5240, {0xc000063020, 0x1, 0x1})
        /home/nic/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:860 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0xca59c0)
        /home/nic/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
        /home/nic/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/exasol/github-keeper/cmd.Execute()
        /home/nic/Projects/github-keeper/cmd/root.go:16 +0x25
main.main()
        /home/user/github-keeper/main.go:6 +0x17
exit status 2
```

--- a/cmd/workflowDefinitionParser.go
+++ b/cmd/workflowDefinitionParser.go
@@ -125,7 +125,7 @@ func (parser WorkflowDefinitionParser) replaceSpecificParameterInJobName(jobName
 	case string:
 		return pattern.ReplaceAllString(jobName, value)
 	case float64:
-		return pattern.ReplaceAllString(jobName, fmt.Sprintf("%f", value))
+		return pattern.ReplaceAllString(jobName, fmt.Sprintf("%.1f", value))
 	case int:
 		return pattern.ReplaceAllString(jobName, fmt.Sprintf("%d", value))
 	case bool:

--- a/cmd/workflowDefinitionParser.go
+++ b/cmd/workflowDefinitionParser.go
@@ -124,6 +124,12 @@ func (parser WorkflowDefinitionParser) replaceSpecificParameterInJobName(jobName
 	switch value := value.(type) {
 	case string:
 		return pattern.ReplaceAllString(jobName, value)
+	case float64:
+		return pattern.ReplaceAllString(jobName, fmt.Sprintf("%f", value))
+	case int:
+		return pattern.ReplaceAllString(jobName, fmt.Sprintf("%d", value))
+	case bool:
+		return pattern.ReplaceAllString(jobName, fmt.Sprintf("%t", value))
 	case map[interface{}]interface{}:
 		filledJobName := jobName
 		for objectKey, objectValue := range value {

--- a/cmd/workflowDefinitionParser_test.go
+++ b/cmd/workflowDefinitionParser_test.go
@@ -331,14 +331,16 @@ func (suite *WorkflowDefinitionParserSuite) TestGetChecksForWorkflowWithFloatPar
       strategy:
         fail-fast: false
         matrix:
-          python: [3.6]
+          python: [3.6, 3.72, 3.86]
           exasol_version:
             - 7.1.6
       name: Run Tests (Python-${{ matrix.python }}, Exasol-${{ matrix.exasol_version }})
 `)
 	suite.NoError(err)
-	suite.Len(definition.JobsNames, 1)
-	suite.Contains(definition.JobsNames, "Run Tests (Python-3.600000, Exasol-7.1.6)")
+	suite.Len(definition.JobsNames, 3)
+	suite.Contains(definition.JobsNames, "Run Tests (Python-3.6, Exasol-7.1.6)")
+	suite.Contains(definition.JobsNames, "Run Tests (Python-3.7, Exasol-7.1.6)")
+	suite.Contains(definition.JobsNames, "Run Tests (Python-3.9, Exasol-7.1.6)")
 }
 
 func (suite *WorkflowDefinitionParserSuite) TestGetChecksForWorkflowWithIntParameter() {

--- a/cmd/workflowDefinitionParser_test.go
+++ b/cmd/workflowDefinitionParser_test.go
@@ -260,12 +260,12 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - python-version: "3.9"
+          - python-version: 3.9
             exasol-tag: latest-7.1
             exasol-port: 8563
 
-          - python-version: "3.9"
-            exasol-tag: latest-7.0
+          - python-version: 3.95
+            exasol-tag: "latest-7.0"
             exasol-port: 8563
 
           - python-version: "3.6"
@@ -276,7 +276,7 @@ jobs:
 	suite.NoError(err)
 	suite.Len(definition.JobsNames, 3)
 	suite.Contains(definition.JobsNames, "Build with Python 3.9 and Exasol latest-7.1")
-	suite.Contains(definition.JobsNames, "Build with Python 3.9 and Exasol latest-7.0")
+	suite.Contains(definition.JobsNames, "Build with Python 3.95 and Exasol latest-7.0")
 	suite.Contains(definition.JobsNames, "Build with Python 3.6 and Exasol latest-6.2")
 }
 

--- a/cmd/workflowDefinitionParser_test.go
+++ b/cmd/workflowDefinitionParser_test.go
@@ -318,3 +318,69 @@ jobs:
 `)
 	suite.Assert().Error(err)
 }
+
+func (suite *WorkflowDefinitionParserSuite) TestGetChecksForWorkflowWithFloatParameter() {
+	parser := WorkflowDefinitionParser{}
+	definition, err := parser.ParseWorkflowDefinition(`
+  name: CI-CD
+  on:
+    push:
+  jobs:
+    run_tests:
+      runs-on: ubuntu-latest
+      strategy:
+        fail-fast: false
+        matrix:
+          python: [3.6]
+          exasol_version:
+            - 7.1.6
+      name: Run Tests (Python-${{ matrix.python }}, Exasol-${{ matrix.exasol_version }})
+`)
+	suite.NoError(err)
+	suite.Len(definition.JobsNames, 1)
+	suite.Contains(definition.JobsNames, "Run Tests (Python-3.600000, Exasol-7.1.6)")
+}
+
+func (suite *WorkflowDefinitionParserSuite) TestGetChecksForWorkflowWithIntParameter() {
+	parser := WorkflowDefinitionParser{}
+	definition, err := parser.ParseWorkflowDefinition(`
+  name: CI-CD
+  on:
+    push:
+  jobs:
+    run_tests:
+      runs-on: ubuntu-latest
+      strategy:
+        fail-fast: false
+        matrix:
+          python: [3]
+          exasol_version:
+            - 7.1.6
+      name: Run Tests (Python-${{ matrix.python }}, Exasol-${{ matrix.exasol_version }})
+`)
+	suite.NoError(err)
+	suite.Len(definition.JobsNames, 1)
+	suite.Contains(definition.JobsNames, "Run Tests (Python-3, Exasol-7.1.6)")
+}
+
+func (suite *WorkflowDefinitionParserSuite) TestGetChecksForWorkflowWithBoolParameter() {
+	parser := WorkflowDefinitionParser{}
+	definition, err := parser.ParseWorkflowDefinition(`
+  name: CI-CD
+  on:
+    push:
+  jobs:
+    run_tests:
+      runs-on: ubuntu-latest
+      strategy:
+        fail-fast: false
+        matrix:
+          success: [true]
+          exasol_version:
+            - 7.1.6
+      name: Run Tests (Success-${{ matrix.success }}, Exasol-${{ matrix.exasol_version }})
+`)
+	suite.NoError(err)
+	suite.Len(definition.JobsNames, 1)
+	suite.Contains(definition.JobsNames, "Run Tests (Success-true, Exasol-7.1.6)")
+}

--- a/doc/changes/changes_0.1.1.md
+++ b/doc/changes/changes_0.1.1.md
@@ -1,7 +1,8 @@
 # Github-Keeper 0.1.1, released 2022-02-??
 
-Code name:
+Code name: Bugfixes
 
 ## Bug Fixes:
 
 * #45: Fixed validation errors for non-pullrequest workflow files
+* #48: Fixed parsing of matrix parameters of types int, float and boolean

--- a/doc/changes/changes_0.1.1.md
+++ b/doc/changes/changes_0.1.1.md
@@ -2,6 +2,8 @@
 
 Code name: Bugfixes
 
+In release 0.1.1 GK now ignores workflow files that are not relevant for pull requests. You can now use floats (`3.7`), integers (`42`) and booleans (`true`) as parameters in a matrix build. Please note that floats will always be rounded to one decimal place, e.g. `3.74` will be rounded to `3.7` and `3.75` will be rounded to `3.8`. If this causes issues with generated branch protection rules, please use strings (`3.75`) as parameters.
+
 ## Bug Fixes:
 
 * #45: Fixed validation errors for non-pullrequest workflow files

